### DISCUSSION TITLE
clear-creds API is not clearing the authorities loaded over VICI

### DISCUSSION THREAD

**Discussion by digambaringale333** [strongswan/strongswan#652] [Q&A] [answered]
Hi,
We are using DAVICI API's in our application to manage IPSec tunnels. In some cases we want to clear all credentials (end point certs and CA authorities ) which are loaded over VICI, for that we are calling clear-creds API but we could see that it's clearing only end point certificates and authorities still remain loaded over VICI.  (** swanctl --list-auth** and **swanctl --list-certs** shows the CA auth even after clear-creds command execution)

Is this expected behavior?

======================
clear-creds()

Clear all loaded certificate, private key and shared key credentials. This affects only credentials loaded over vici, but additionally flushes the credential cache.

{} => {
    success = <yes or no>
    errmsg = <error string on failure>
}
=================

**Comment by tobiasbrunner** [comment, marked as answer]
> Is this expected behavior?

Yes. The `clear-creds` vici command only clears credentials loaded via `load-cert`, `load-key` and `load-shared`. If you loaded an authority via `load-authority` command, you have to unload it using the `unload-authority` command.

**Reply by digambaringale333** [reply]
Thanks for your quick reply Tobias, I really appreciate it!
If there are multiple authorities loaded, do we have to unload one authority at at time (because unload-authority takes one auth handle at a time)?
